server/hotrod/src/main/java/org/infinispan/server/hotrod/configuration/HotRodServerConfiguration.java

```diff
@@ -1,69 +1,83 @@
+   public static final AttributeDefinition<Boolean> TOPOLOGY_STATE_TRANSFER = AttributeDefinition.builder("topology-state-transfer", true).immutable().build();
+
+   private final Attribute<String> proxyHost;
+   private final Attribute<Integer> proxyPort;
+   private final Attribute<Long> topologyLockTimeout;
+   private final Attribute<Long> topologyReplTimeout;
+   private final Attribute<Boolean> topologyAwaitInitialTransfer;
+   private final Attribute<Boolean> topologyStateTransfer;
    private final AuthenticationConfiguration authentication;
 
-   HotRodServerConfiguration(String defaultCacheName, String proxyHost, int proxyPort,
-                             long topologyLockTimeout, long topologyReplTimeout, boolean topologyAwaitInitialTransfer, boolean topologyStateTransfer,
-                             String name, String host, int port,
-                             int idleTimeout, int recvBufSize, int sendBufSize, SslConfiguration ssl, boolean tcpNoDelay, boolean tcpKeepAlive,
-                             int workerThreads, AuthenticationConfiguration authentication, Set<String> ignoredCaches,
-                             boolean startTransport, AdminOperationsHandler adminOperationsHandler) {
-      super(defaultCacheName, name, host, port,
-            idleTimeout, recvBufSize, sendBufSize, ssl, tcpNoDelay, tcpKeepAlive, workerThreads,
-            ignoredCaches, startTransport, adminOperationsHandler);
-      this.proxyHost = proxyHost;
-      this.proxyPort = proxyPort;
-      this.topologyCacheName = TOPOLOGY_CACHE_NAME_PREFIX + (name.length() > 0 ? "_" + name : name);
-      this.topologyLockTimeout = topologyLockTimeout;
-      this.topologyReplTimeout = topologyReplTimeout;
-      this.topologyStateTransfer = topologyStateTransfer;
-      this.topologyAwaitInitialTransfer = topologyAwaitInitialTransfer;
+   public static AttributeSet attributeDefinitionSet() {
+      return new AttributeSet(HotRodServerConfiguration.class, ProtocolServerConfiguration.attributeDefinitionSet(),
+            PROXY_HOST, PROXY_PORT, TOPOLOGY_STATE_TRANSFER, TOPOLOGY_AWAIT_INITIAL_TRANSFER, TOPOLOGY_LOCK_TIMEOUT, TOPOLOGY_REPL_TIMEOUT);
+   }
+
+   HotRodServerConfiguration(AttributeSet attributes, SslConfiguration ssl, AuthenticationConfiguration authentication) {
+      super(attributes, ssl);
       this.authentication = authentication;
+
+      proxyHost = attributes.attribute(PROXY_HOST);
+      proxyPort = attributes.attribute(PROXY_PORT);
+      topologyLockTimeout = attributes.attribute(TOPOLOGY_LOCK_TIMEOUT);
+      topologyReplTimeout = attributes.attribute(TOPOLOGY_REPL_TIMEOUT);
+      topologyAwaitInitialTransfer = attributes.attribute(TOPOLOGY_AWAIT_INITIAL_TRANSFER);
+      topologyStateTransfer = attributes.attribute(TOPOLOGY_STATE_TRANSFER);
    }
 
    public String proxyHost() {
-      return proxyHost;
+      return proxyHost.get();
+   }
+
+   public String actualHost() {
```

Comment:
`actualHost` isn't the best name IMO... `publicHost`? `exposedHost`?

Reply:
publicHost is good